src/query/validation.c

```diff
@@ -231,10 +231,11 @@ static void verify_bucket_expression(Node *node)
   if (IsA(node, FuncExpr))
   {
     FuncExpr *func_expr = (FuncExpr *)node;
-    if (is_allowed_cast(func_expr->funcid))
+    if (is_allowed_cast(func_expr))
     {
       Assert(list_length(func_expr->args) == 1); /* All allowed casts require exactly one argument. */
       verify_bucket_expression(linitial(func_expr->args));
+      return;
```

Comment:
This might seem like a pretty big change, but I was surprised it didn't return. When a function expression is an allowed cast, it makes sense to validate the argument and not proceed with the usual validation. Common sense points me to this being right now, am I missing something?

Reply:
No, makes sense to me too!